### PR TITLE
CTE produce output optimize

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Add output columns on sender to reduce CTE produce output columns copy

```
tpcds_100g_performance>  with xx as (select * from store_sales)
                     -> select *
                     -> from (select count(distinct xx.ss_sold_date_sk) from  xx) t1,
                     ->      (select count(distinct xx.ss_sold_time_sk) from  xx) t2,
                     ->      (select count(distinct xx.ss_promo_sk) from  xx) t3,
                     ->      (select count(distinct xx.ss_customer_sk) from  xx) t4,
                     ->      (select count(distinct xx.ss_cdemo_sk) from  xx) t5,
                     ->      (select count(distinct xx.ss_wholesale_cost) from  xx) t6,
                     ->      (select count(distinct xx.ss_list_price) from  xx) t7,
                     ->      (select count(distinct xx.ss_net_paid) from  xx) t8,
                     ->      (select count(distinct xx.ss_net_profit) from  xx) t9;
```

Before: Time: 4.439s
After: Time: 3.037s

### CODE REVIEW

File: be/src/runtime/data_stream_sender.cpp [hunk 1]
```diff
@@ -386,7 +386,8 @@ DataStreamSender::DataStreamSender(RuntimeState* state, int sender_id, const Row
           _bytes_sent_counter(nullptr),
           _dest_node_id(sink.dest_node_id),
           _destinations(destinations),
-          _enable_exchange_pass_through(enable_exchange_pass_through) {
+          _enable_exchange_pass_through(enable_exchange_pass_through),
+          _output_columns(sink.output_columns) {
```
Comment (dirtysalt) [inline be/src/runtime/data_stream_sender.cpp:390]: Maybe this is non-pipeline version? Maybe we don't need it.

**Review by dirtysalt — DISMISSED** [review]

File: be/src/runtime/data_stream_sender.cpp (hunk 1 above)
Comment (Seaven) [reply be/src/runtime/data_stream_sender.cpp:390]: Yeah, you are right, don't need modify it

**Comment by Seaven** [comment]
run all

File: be/src/runtime/data_stream_sender.cpp [hunk 2]
```diff
@@ -549,13 +550,25 @@ Status DataStreamSender::send_chunk(RuntimeState* state, vectorized::Chunk* chun
     if (num_rows == 0) {
         return Status::OK();
     }
+
+    vectorized::Chunk send_chunk_tmp;
```
Comment (kangkaisen) [inline be/src/runtime/data_stream_sender.cpp:554]: needn't change here. only need to update pipeline engine codes

File: be/src/exec/pipeline/exchange/exchange_sink_operator.h [hunk 3]
```diff
@@ -177,6 +180,8 @@ class ExchangeSinkOperatorFactory final : public OperatorFactory {
     bool _enable_exchange_pass_through;
 
     FragmentContext* const _fragment_ctx;
+
+    const std::vector<int32_t>& _output_columns;
```
Comment (kangkaisen) [inline be/src/exec/pipeline/exchange/exchange_sink_operator.h:184]: _output_columns should be value in ExchangeSinkOperatorFactory, and reference in ExchangeSinkOperator

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 31 / 32 (96.88%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/DataStreamSink.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|18|18|100.00%|
|:large_blue_circle:|com/starrocks/planner/ExchangeNode.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastDataSink.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastPlanFragment.java|1|1|100.00%|

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 31 / 32 (96.88%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/DataStreamSink.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|18|18|100.00%|
|:large_blue_circle:|com/starrocks/planner/ExchangeNode.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastDataSink.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastPlanFragment.java|1|1|100.00%|

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 31 / 32 (96.88%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/DataStreamSink.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|18|18|100.00%|
|:large_blue_circle:|com/starrocks/planner/ExchangeNode.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastDataSink.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastPlanFragment.java|1|1|100.00%|

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 31 / 32 (96.88%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/DataStreamSink.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|18|18|100.00%|
|:large_blue_circle:|com/starrocks/planner/ExchangeNode.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastDataSink.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/planner/MultiCastPlanFragment.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**